ODIN_II/SRC/netlist_check.cpp

```diff
@@ -46,11 +46,15 @@ void depth_first_traverse_check_if_forward_leveled(nnode_t* node, uintptr_t trav
 void sequential_levelized_dfs(short marker_value, netlist_t* netlist);
 void depth_first_traverse_until_next_ff_or_output(nnode_t* node, nnode_t* calling_node, uintptr_t traverse_mark_number, int seq_level, netlist_t* netlist);
 
+void check_discontinuity(netlist_t* netlist);
+void discontinuity_check_forward_traverse(nnode_t* node, operation_list op, long long& num_of_forward_elements, unsigned short mark);
+void discontinuity_check_backward_traverse(nnode_t* node, operation_list op, long long& num_of_backward_elements, unsigned short mark);
 /*---------------------------------------------------------------------------------------------
  * (function: check_netlist)
  * Note: netlist passed in needs to be initialized by allocate_netlist() to make sure correctly initialized.
  *-------------------------------------------------------------------------------------------*/
 void check_netlist(netlist_t* netlist) {
+    check_discontinuity(netlist);
```

Comment:
Did we have the same process in the `netlist_statistics.cpp` file? I am pretty sure the procedure is almost the same, please have a look at that file and add warnings when we traverse a netlist in that file